src/readers/src/main/scala/KeyValueReaderIterator.scala

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
```

Comment:
I'm still a bit concerned that we have to copy-paste ~50 lines from Spark in order to return a tuple instead of a single value.
In any event, I would suggest renaming the type params K and V instead of K and T since that seems more idiomatic to code I've seen.

Reply:
done